packages/client/tests/functional/referentialActions-setDefault/prisma/_schema.ts

```diff
@@ -0,0 +1,37 @@
+import { match } from 'ts-pattern'
+
+import { Providers } from '../../_utils/providers'
+import testMatrix from '../_matrix'
+import { schema_1to1 } from './_schema_1_to_1'
+import { schema_1ton } from './_schema_1_to_n'
+
+// Note: We can't test `SetDefault` with automatically created ids in SQLServer.
+// Had we defined Prisma models with automatic ids in the form of `id Int @id @default(autoincrement())`,
+// SQL Server would have thrown a runtime error at the `CREATE TABLE` level, as Prisma would use the IDENTITY(1, 1) type
+// for the id column, which cannot be created or updated explicitly.
+// The escape hatch would be using `SET IDENTITY_INSERT OFF`, which is however allowed only for a single table per session,
+// and isn't supported by Prisma (not even in raw mode, see https://github.com/prisma/prisma/issues/15305)
+//
+// The obvious solution for us was to avoid `@default(autoincrement())` in tests.
+// Providing a separate `id` schema definition for each provider is not necessary.
```

Comment:
This is something I remember bumping into a few weeks ago, and found again while working on the tests. I'm open to move this text somewhere else

Reply:
In this case, this is not an issue of the referential actions, and we don't need to differentiate a schema based on the database for this test. The test we wrote is not a "worse test" compared to what we tested on other databases.

Rather, the comment highlighted above indicates that Prisma users fond of automatically generate ids on SQLServer will have a hard time triggering a referential action in practice. This is a DX issue unrelated to referential integrity.

Reply:
TODO after merge: open a new issue about this